File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        dummyHead = ListNode()
        c1, c2, cdh = list1, list2, dummyHead
        while c1 and c2:
            if c1.val <= c2.val:
                cdh.next = c1
                c1 = c1.next
            else:
                cdh.next = c2
                c2 = c2.next
            cdh = cdh.next
        
        if c1:
            cdh.next = c1
        else:
            cdh.next = c2
            
        return dummyHead.next

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
  dh, c1, c2 := ListNode{}, list1, list2
  cdh := &dh
  
  for c1 != nil && c2 != nil {
    if c1.Val <= c2.Val {
      cdh.Next = c1
      c1 = c1.Next
    } else {
      cdh.Next = c2
      c2 = c2.Next
    }
    cdh = cdh.Next
  }
  if c1 != nil {
    cdh.Next = c1
  } else {
    cdh.Next = c2
  }
  return dh.Next
}